### PR TITLE
frr: support rpki

### PR DESCRIPTION
**Pull request by Anillc**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/frr.nix [hunk 1]
```diff
@@ -106,6 +106,14 @@ let
           TCP Port to bind to for the VTY interface.
         '';
       };
+
+      extraOptions = mkOption {
+          type = types.str;
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/frr.nix:111]: This should be an attr of strings to easily supply multiple options.

Comment (SuperSandro2000) [reply nixos/modules/services/networking/frr.nix:111]: hmmm, not sure

Comment (aanderse) [reply nixos/modules/services/networking/frr.nix:111]: Yeah... maybe `with types; listOf str;`.

Comment (Anillc) [reply nixos/modules/services/networking/frr.nix]: Updated to list of strings

**Comment by aanderse** [comment]
Module changes look fine. Since you have to rebase anyways can you please change the spacing in your diff to be a consistent 2 spaces per level of indentation?

**Comment by Anillc** [comment]
> Module changes look fine. Since you have to rebase anyways can you please change the spacing in your diff to be a consistent 2 spaces per level of indentation?

resolved

**Pull request merged**